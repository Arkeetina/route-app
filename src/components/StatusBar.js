import React, { Component } from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import uuid from 'uuid';

import StatusBarText from './StatusBarText';

class StatusBar extends Component {
  loaderRender = () => {
    const { loading } = this.props;
    if (loading) {
      return <img className="loader__image" src="/images/loader.gif"/>;
    }
  }

  defaultMsgRender = () => {
    const { defaultMsg } = this.props;
    if (defaultMsg) {
      return <StatusBarText message={defaultMsg}/>;
    }
  }

  routeInProgStatusRender = () => {
    const { routeInProgStatus } = this.props;
    if (routeInProgStatus) {
      return <StatusBarText message={routeInProgStatus}/>;
    }
  }

  routeFailStatusRender = () => {
    const { routeFailStatus } = this.props;
    if (routeFailStatus) {
      return <StatusBarText message={routeFailStatus}/>;
    }
  }

  serverErrorRender = () => {
    const { serverError } = this.props;
    const message = "Connection Error, please try again later";

    if (serverError) {
      return <StatusBarText message={message}/>;
    }
  }

  pathRender = () => {
    const { totalTime, totalDistance } = this.props;
    if (totalDistance && totalTime) {
      return <StatusBarText distance={totalDistance} time={totalTime}/>;
    }
  }
  
  render() {
    return (
      <div key={uuid()} className="status-bar status-bar__content">
        <ReactCSSTransitionGroup
          transitionName="anim"
          transitionEnterTimeout={500}
          transitionAppear
          transitionAppearTimeout={1000}
          transitionLeaveTimeout={1000}
        >
          {this.defaultMsgRender()}
          {this.routeInProgStatusRender()}
          {this.routeFailStatusRender()}
          {this.serverErrorRender()}
          {this.pathRender()}
          {this.loaderRender()}
        </ReactCSSTransitionGroup>
      </div>
    );
  }
}

export default StatusBar;
