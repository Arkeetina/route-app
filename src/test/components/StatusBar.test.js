import React from 'react';
import { shallow } from 'enzyme';
import StatusBar from '../../components/StatusBar';

test('should render StatusBar correctly', () => {
  const wrapper = shallow(<StatusBar />);
  expect(wrapper).toMatchSnapshot();
});
<<<<<<< HEAD
=======

test('should render StatusBar with default message correctly', () => {
  const defautMsg = 'default message';
  const wrapper = shallow(<StatusBar defaultMsg={defautMsg} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render StatusBar with route fail status correctly', () => {
  const routeFailStatus = 'route fail';
  const wrapper = shallow(<StatusBar routeFailStatus={routeFailStatus} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render StatusBar with route in progress status correctly', () => {
  const routeInProgStatus = 'in progress';
  const wrapper = shallow(<StatusBar routeInProgStatus={routeInProgStatus} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render StatusBar with server error correctly', () => {
  const serverError = 'server error';
  const wrapper = shallow(<StatusBar serverError={serverError} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render StatusBar with total time correctly', () => {
  const totalTime = 9000;
  const wrapper = shallow(<StatusBar totalTime={totalTime} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render StatusBar with total distance correctly', () => {
  const totalDistance = 10000;
  const wrapper = shallow(<StatusBar totalDistance={totalDistance} />);
  expect(wrapper).toMatchSnapshot();
});
>>>>>>> dev1
