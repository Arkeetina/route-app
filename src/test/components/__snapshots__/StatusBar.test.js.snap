// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render StatusBar correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="fe34bd9e-d134-4ba0-9141-87e9788d30b8"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  />
</div>
`;

exports[`should render StatusBar with default message correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="a65809db-8125-438e-900f-89333e69a4f9"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  >
    <StatusBarText
      message="default message"
    />
  </CSSTransitionGroup>
</div>
`;

exports[`should render StatusBar with route fail status correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="c6c8f94f-5b78-4435-b7e6-4526b3e09e95"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  >
    <StatusBarText
      message="route fail"
    />
  </CSSTransitionGroup>
</div>
`;

exports[`should render StatusBar with route in progress status correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="ffe35c91-b81c-4034-ab37-4dd1861ffb28"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  >
    <StatusBarText
      message="in progress"
    />
  </CSSTransitionGroup>
</div>
`;

exports[`should render StatusBar with server error correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="8c1ab9ae-1738-40af-b8b0-fe781c2f7849"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  >
    <StatusBarText
      message="Connection Error, please try again later"
    />
  </CSSTransitionGroup>
</div>
`;

exports[`should render StatusBar with total distance correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="00022d42-367c-4cd9-a943-237cad6ec02a"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  />
</div>
`;

exports[`should render StatusBar with total time correctly 1`] = `
<div
  className="status-bar status-bar__content"
  key="5fbbe8d7-eb71-4860-b45b-868ab267aba3"
>
  <CSSTransitionGroup
    transitionAppear={true}
    transitionAppearTimeout={1000}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={1000}
    transitionName="anim"
  />
</div>
`;
